pkg/sbproxy/reconcile/reconcile_brokers.go

```diff
@@ -136,49 +139,111 @@ func (r *resyncJob) getBrokersFromSM(ctx context.Context) ([]*platform.ServiceBr
 	return brokersFromSM, nil
 }
 
-func (r *resyncJob) fetchBrokerCatalog(ctx context.Context, broker *platform.ServiceBroker) error {
+func (r *resyncJob) fetchBrokerCatalog(ctx context.Context, brokerGUIDInPlatform string, brokerInSM *platform.ServiceBroker) error {
 	if f, isFetcher := r.platformClient.(platform.CatalogFetcher); isFetcher {
 		logger := log.C(ctx)
-		logger.WithFields(logBroker(broker)).Infof("resyncJob refetching catalog for broker...")
-		if err := f.Fetch(ctx, broker); err != nil {
-			logger.WithFields(logBroker(broker)).WithError(err).Error("Error during fetching catalog...")
+		logger.WithFields(logBroker(brokerInSM)).Infof("resyncJob refetching catalog for broker...")
+
+		username, password, passwordHash, err := util.GenerateBrokerPlatformCredentials()
+		if err != nil {
+			return fmt.Errorf("could not generate broker platform credentials for broker (%s): %s", brokerInSM.Name, err)
+		}
+
+		credentials := &types.BrokerPlatformCredential{
+			Username:     username,
+			PasswordHash: passwordHash,
+			BrokerID:     brokerInSM.GUID,
+		}
+
+		if err := r.smClient.PutCredentials(ctx, credentials); err != nil {
```

Comment:
i thought full resync should not rotate credentials ? Also how does this work  - i thought the sm api for credentials does not accept empty notification id - https://github.com/Peripli/service-manager/pull/443/files#diff-d14e0868cba782bb989a5e170d599d21R116

same comment applies for `createBrokerRegistration` and `updateBrokerRegistration`